overseer/src/lib.rs

```diff
@@ -299,6 +345,15 @@ where
 			// that need dispatching (if any).
 			let mut msgs = Vec::default();
 
+			while let Poll::Ready(Some(msg)) = poll!(&mut self.events_rx.next()) {
+				if let Event::Stop = msg {
+					// TODO: We should send stop messages to all subsystems, join them
+					// and wait for some timeout for them to gracefully shutdown and then
+					// just drop their handlers.
```

Comment:
It would be nice to have a `cancellation` utility like there is for oneshot, so we could wait for all of the receivers to hang up.

Reply:
Should this be considered as a `TODO [now]` (this PR) or a `TODO [later]` some other PR? If the second case, we will need an issue filed.